src/ui/qt/qt_ui.cpp

```diff
@@ -44,3 +44,25 @@ void UiWindowQt::drawLine( const Point2d & start, const Point2d & end, const Pai
     _window.setPixmap( _pixmap );
     _display();
 }
+
+void UiWindowQt::drawEllipse( const Point2d & center, double xDiameter, double yDiameter, const PaintColor & color )
+{
+    QPainter paint( &_pixmap );
+    paint.setPen( QColor( color.red, color.green, color.blue, color.alpha ) );
+    paint.drawEllipse( static_cast<int>(center.x - (xDiameter / 2)), static_cast<int>(center.y - (yDiameter / 2)),
```

Comment:
Why don't we use `drawEllipse(const QPointF & center, qreal rx, qreal ry)` instead? It seems more reasonable. Plus we can use **radius** instead of **diameter** which brings more sense.

Reply:
I'm ok with radius..
But regarding the datatype QPointF and qreal, I'm thinking:
This is Qt specific data types which is fine for qt_ui but how will the base class be affected if we use these datatypes? Will it be fine to use different datatypes in virtual functions in our base class and the derived class?

Reply:
My apology for late reply. What I meant is to change only Qt drawEllipse function without changing our function interface. We can do casting from double to float if QPointF is float based class/structure. Leaving accuracy in such case is not a big problem as drawings are for human eye and person cannot recognise very small subpixel shift.

Reply:
`QPointF` accepts two `qreal` variables which are `double` so our code would look like:
```cpp
paint.drawEllipse( QPointF(center.x, center.y), xRadius, yRadius );
```
Without any changes in our function's input parameter list :)

Reply:
Thanks!
I'll implement this in my next commit 👍